lib/azure/armrest/model/storage_account.rb

```diff
@@ -119,14 +119,57 @@ def table_data(name, key = nil, options = {})
       # If no key is provided, it is assumed that the StorageAccount object
       # includes the key1 property.
       #
-      def containers(key = nil)
+      # # The following options are supported:
+      #
+      # * prefix
+      # * delimiter
+      # * maxresults
+      # * include
+      # * timeout
+      #
+      # By default Azure uses a value of 5000 for :maxresults.
+      #
+      # If the :include option is specified, it should contain an array of
+      # one element: metadata. More options may be added by Microsoft
+      # at a later date.
+      #
+      # Example:
+      #
+      #   sas  = Azure::Armrest::StorageAccountService.new(conf)
+      #   key  = sas.list_account_keys['key1']
+      #   acct = sas.get('your_storage_account', 'your_resource_group')
+      #
+      #   p acct.containers(key)
+      #   p acct.containers(key, :include => ['metadata'])
+      #   p acct.containers(key, :maxresults => 1)
+      #
+      # In cases where a NextMarker element is found in the original response,
+      # another call will automatically be made with the marker value included
+      # in the URL so that you don't have to perform such a step manually.
+      #
+      def containers(key = nil, options = {})
         key ||= properties.key1
 
-        response = blob_response(key, "comp=list")
+        query = "comp=list"
+        options.each { |okey, ovalue| query += "&#{okey}=#{[ovalue].flatten.join(',')}" }
+
+        response = blob_response(key, query)
 
-        Nokogiri::XML(response.body).xpath('//Containers/Container').map do |element|
+        doc = Nokogiri::XML(response.body)
+
+        results = doc.xpath('//Containers/Container').collect do |element|
           Container.new(Hash.from_xml(element.to_s)['Container'])
         end
+
+        doc.xpath('//NextMarker').each do |xmarker|
```

Comment:
is it possible to pull this code into a reusable method? the logic is the same for containers and blobs